service/cluster/xtra_db_cluster.go

```diff
@@ -80,38 +76,62 @@ func (s *XtraDBClusterService) ListXtraDBClusters(ctx context.Context, req *cont
 			Pxc: &controllerv1beta1.XtraDBClusterParams_PXC{
 				DiskSize: int64(pxcDiskSize),
 			},
-			Proxysql: &controllerv1beta1.XtraDBClusterParams_ProxySQL{
-				DiskSize: int64(proxySQLDiskSize),
-			},
 		}
-		if cluster.PXC.ComputeResources != nil {
-			cpuMillis, err := convertors.StrToMilliCPU(cluster.PXC.ComputeResources.CPUM)
+
+		if cluster.ProxySQL != nil {
+			proxySQLDiskSize, err := convertors.StrToBytes(cluster.ProxySQL.DiskSize)
 			if err != nil {
 				return nil, status.Error(codes.Internal, err.Error())
 			}
-			memoryBytes, err := convertors.StrToBytes(cluster.PXC.ComputeResources.MemoryBytes)
+			params.Proxysql = &controllerv1beta1.XtraDBClusterParams_ProxySQL{
+				DiskSize: int64(proxySQLDiskSize),
+			}
+			if cluster.ProxySQL.ComputeResources != nil {
+				cpuMillis, err := convertors.StrToMilliCPU(cluster.ProxySQL.ComputeResources.CPUM)
+				if err != nil {
+					return nil, status.Error(codes.Internal, err.Error())
+				}
+				memoryBytes, err := convertors.StrToBytes(cluster.ProxySQL.ComputeResources.MemoryBytes)
+				if err != nil {
+					return nil, status.Error(codes.Internal, err.Error())
+				}
+				params.Proxysql.ComputeResources = &controllerv1beta1.ComputeResources{
+					CpuM:        int32(cpuMillis),
+					MemoryBytes: int64(memoryBytes),
+				}
+			}
+		} else if cluster.HAProxy.ComputeResources != nil {
```

Comment:
maybe we should not use `else if` if there is no `else` at the end, just  use two seperate if conditions

Reply:
:+1: Done.